Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameProxy.cpp

```diff
@@ -127,6 +127,12 @@ CVPixelBufferRef RemoteVideoFrameProxy::pixelBuffer() const
         m_pixelBuffer = WebCore::createBlackPixelBuffer(static_cast<size_t>(m_size.width()), static_cast<size_t>(m_size.height()));
     return m_pixelBuffer.get();
 }
+
+VideoFrame::ResourceIdentifier RemoteVideoFrameProxy::resourceIdentifier() const
+{
+    auto ref = newReadReference();
```

Comment:
This is potentially dangerous as creating a read reference increments the pending read counter.
When creating such a read reference, we should ensure to send the read reference to GPUProcess through IPC.
It is difficult to me to tell whether the new code path actually ensures this.
In any case, resourceIdentifier() makes it as if it is without side effect, which is not the case.
We should at least find a better name for this routine. If we use SharedVideoFrame and a single IPC message, we might be able to remove this routine.

This makes me wonder whether we should not revisit the current remote video frame strategy, given this and other stuff (some desire to create some remote video frame in UIProcess, desire to move some video frames from one web process to another).

Reply:
Maybe I can use `SharedVideoFrameWriter` instead since I think this is achieving the same thing, I will attempt to refactor this.